### PR TITLE
feat: added rail_fence_cipher.cpp

### PR DESCRIPTION
**Pull request by sushmit-goyal**
#### Description of Change
Added rail_fence_cipher.cpp

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1267"></a>

### CODE REVIEW

**Review by kvedala — CHANGES_REQUESTED** [review]
👍 
good code. note the comments

File: ciphers/rail_fence_cipher.cpp [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+         std::string Cipher::rail_fence_encrypt(std::string text, int key){
+
+             
+            int num_rows = key;
+            int num_columns = text.length();
+
+            // create the matrix to cipher plain text
+            // key = rows , length(text) = columns
+            std::vector<std::string> rail(num_rows);
```
[the remark this answers is not in the document]
Comment (kvedala) [reply ciphers/rail_fence_cipher.cpp:92]: Try for example: changing the test case "ILoveAlgorithms" to "I Love Algorithms".

[the remark this answers is not in the document]
Comment (sushmit-goyal) [reply ciphers/rail_fence_cipher.cpp]: Changed the case to "I Love Algorithms". Seems to work fine. Please have a look now

File: ciphers/rail_fence_cipher.cpp [hunk 2]
```diff
@@ -0,0 +1,274 @@
+/*
```
Comment (kvedala) [inline ciphers/rail_fence_cipher.cpp:1]:
```suggestion
/**
```

doc-comment blocks must begin with `/**` and then with `*/`. Please update other locations as well.

**Pull request closed without merging**